core/domain/blog_statistics_services.py

```diff
@@ -0,0 +1,1094 @@
+
+    Args:
+        author_id: str. User ID of the author.
+
+    Returns:
+        AuthorBlogPostsReadingTime. The author blog post reading time stats
+        domain object.
+    """
+    reading_time_stats = (
+        blog_stats_models.AuthorBlogPostAggregatedReadingTimeModel.get(
+            author_id
+        )
+    )
+    return blog_statistics_domain.AuthorBlogPostsReadingTime(
+        author_id,
+        reading_time_stats.zero_to_one_min,
+        reading_time_stats.one_to_two_min,
+        reading_time_stats.two_to_three_min,
+        reading_time_stats.three_to_four_min,
+        reading_time_stats.four_to_five_min,
+        reading_time_stats.five_to_six_min,
+        reading_time_stats.six_to_seven_min,
+        reading_time_stats.seven_to_eight_min,
+        reading_time_stats.eight_to_nine_min,
+        reading_time_stats.nine_to_ten_min,
+        reading_time_stats.more_than_ten_min,
+    )
+
+
+def save_blog_post_views_stats_model(
+    views_stats: blog_statistics_domain.BlogPostViewsAggregatedStats
+) -> None:
+    """Updates the BlogPostViewsAggregatedStatsModel datastore instance with the
+    passed BlogPostViewsAggregatedStats domain object.
+
+    Args:
+        views_stats: BlogPostViewsAggregatedStats. The blog post views
+            aggregated stats domain object.
+
+    Raises:
+        Exception. No blog post views stats model exists for the given
+            blog_post_id.
+    """
+    blog_post_views_aggregated_stats = (
+        blog_stats_models.BlogPostViewsAggregatedStatsModel.get(
+            views_stats.blog_post_id, strict=False
+        )
+    )
+    if blog_post_views_aggregated_stats is None:
+        raise Exception(
+            'No blog post views stats model exists for the given blog_post_id.'
+        )
+
+    blog_post_views_aggregated_stats.views_by_hour = views_stats.views_by_hour
+    blog_post_views_aggregated_stats.views_by_month = views_stats.views_by_month
+    blog_post_views_aggregated_stats.views_by_date = views_stats.views_by_date
+
+    blog_post_views_aggregated_stats.update_timestamps()
+    blog_post_views_aggregated_stats.put()
+
+
+def save_blog_post_reads_stats_model(
+    reads_stats: blog_statistics_domain.BlogPostReadsAggregatedStats
+) -> None:
+    """Updates the BlogPostReadsAggregatedStatsModel datastore instance with the
+    passed BlogPostReadsAggregatedStats domain object.
+
+    Args:
+        reads_stats: BlogPostReadsAggregatedStats. The blog post reads
+            aggregated stats domain object.
+
+    Raises:
+        Exception. No blog post reads stats model exists for the given
+            blog_post_id.
+    """
+    blog_post_reads_aggregated_stats = (
+        blog_stats_models.BlogPostReadsAggregatedStatsModel.get(
+            reads_stats.blog_post_id, strict=False
+        )
+    )
+    if blog_post_reads_aggregated_stats is None:
+        raise Exception(
+            'No blog post reads stats model exists for the given blog_post_id.'
+        )
```

Comment:
Same as above, please include blog_post_id in the error message as well.

Reply:
DOne